app/controllers/events_controller.rb

```diff
@@ -40,6 +40,9 @@ def update
 
     if @event.alpacas << @alpaca
       @points = CalculateEventPointsAndAssignToAlpacas.new(@event).call
+      @winner = Alpaca.find(@points.max_by{ |_alpaca_id, points| points }.first)
```

Comment:
* `@points.max_by{ |_alpaca_id, points| points }` is used twice in this code, so maybe it should be assigned to local variable?
* `update_attributes` skips validation. We should always use `update`, unless there is a specific problem, which requires skipping validation

Reply:
1 - yes it could, and even should ;) I just tried to push it as fast as it was possible
2 - update_attributes != update_attribute